src/Imgix/UrlBuilder.cs

```diff
@@ -56,40 +55,217 @@ public String BuildUrl(String path)
             return BuildUrl(path, new Dictionary<string, string>());
         }
 
+        /// <summary>
+        /// Create a resolvable imgix url.
+        /// 
+        /// </summary>
+        /// <param name="path">path to the image, i.e. "image/file.png"</param>
+        /// <param name="parameters">dictionary of query parameters</param>
+        /// <returns>resolvable URL</returns>
+
+        /*
+        * Encodes the `path` and `paramater` arguments to the UTF8 
+        * scheme and returns a formatted string that conforms to the 
+        * "{scheme}://{domain}/{path}{params}" format.
+        */
         public String BuildUrl(String path, Dictionary<String, String> parameters)
         {
+            String scheme = UseHttps ? "https" : "http";
+            path = SanitizePath(path);
+
+            var qs = BuildParams(parameters);
+            var localParams = new Dictionary<String, String>(parameters);
+
+            if (!String.IsNullOrEmpty(_signKey))
+            {
+                var hashString = String.Format("{0}/{1}{2}", _signKey, path, localParams.Any() ? "?" + qs : String.Empty);
+                localParams.Add("s", HashString(hashString));
```

Comment:
I know it's out of scope for this PR, but I also think we should rename the `HashString` method to more closely match [js-core](https://github.com/imgix/js-core/blob/main/src/index.js#L83-L90)